src/proxy/createClassProxy.js

```diff
@@ -225,40 +225,47 @@ function createClassProxy(InitialComponent, proxyKey, options) {
     // This function only gets called for the initial mount. The actual
     // rendered component instance will be the return value.
 
-    // eslint-disable-next-line func-names
-    ProxyFacade = function(props, context) {
-      const result = CurrentComponent(props, context)
-
-      // simple SFC
-      if (!CurrentComponent.contextTypes) {
-        if (!ProxyFacade.isStatelessFunctionalProxy) {
-          setSFPFlag(ProxyFacade, true)
-        }
-
-        return renderOptions.componentDidRender(result)
-      }
-      setSFPFlag(ProxyFacade, false)
-
-      // This is a Relay-style container constructor. We can't do the prototype-
-      // style wrapping for this as we do elsewhere, so just we just pass it
-      // through as-is.
-      if (isReactComponentInstance(result)) {
-        ProxyComponent = null
-        return result
-      }
-
-      // Otherwise, it's a normal functional component. Build the real proxy
-      // and use it going forward.
+    if (1) {
```

Comment:
SFC have to be wrapped by "Components", or Preact will wrap them. (to be controlled via setConfig)